src/diagnostics/core_diagnostics.jl

```diff
@@ -1201,6 +1201,67 @@ add_diagnostic_variable!(
     compute! = compute_clvi!,
 )
 
+###
+# Cloud top height
+###
+compute_cloud_top_height!(out, state, cache, time) = compute_cloud_top_height!(
+    out,
+    state,
+    cache,
+    time,
+    cache.atmos.moisture_model,
+)
+compute_cloud_top_height!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("cltz", model)
+
+#heaviside function version
+function compute_cloud_top_height!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+
+    clw = @. lazy(state.c.ρ * cache.precomputed.cloud_diagnostics_tuple.q_liq)
+    cli = @. lazy(state.c.ρ * cache.precomputed.cloud_diagnostics_tuple.q_ice)
+    z = Fields.coordinate_field(state.c).z
+
+    FT = Spaces.undertype(axes(clw))
+
+    q_cond = @. lazy(clw + cli)
+
+    heaviside_num =
```

Comment:
It's not really a Heaviside function. Better call it `cloud_mask` or something like it (with the modifications below).